contracts/misc/AutoYieldApe.sol

```diff
@@ -0,0 +1,357 @@
+        external
+        view
+        override
+        returns (
+            address,
+            address,
+            uint256
+        )
+    {
+        return (_yieldUnderlying, _yieldToken, _currentYieldIndex);
+    }
+
+    function setHarvestOperator(address _harvestOperator) external onlyOwner {
+        require(_harvestOperator != address(0), "zero address");
+        address oldOperator = harvestOperator;
+        if (oldOperator != _harvestOperator) {
+            harvestOperator = _harvestOperator;
+            emit HarvestOperatorUpdated(oldOperator, _harvestOperator);
+        }
+    }
+
+    function setHarvestFee(uint256 _harvestFee) external onlyOwner {
+        require(
+            _harvestFee < PercentageMath.HALF_PERCENTAGE_FACTOR,
+            "Fee Too High"
+        );
+        uint256 oldValue = harvestFee;
+        if (oldValue != _harvestFee) {
+            harvestFee = _harvestFee;
+            emit HarvestFeeUpdated(oldValue, _harvestFee);
+        }
+    }
+
+    function rescueERC20(
+        address token,
+        address to,
+        uint256 amount
+    ) external onlyOwner {
+        require(token != address(_yieldToken), "cannot rescue yield token");
+        IERC20(token).safeTransfer(to, amount);
+        emit RescueERC20(token, to, amount);
+    }
+
+    function _withdraw(uint256 amount) internal {
+        require(amount > 0, "zero amount");
+
+        _updateYieldIndex(msg.sender);
+        _burn(msg.sender, amount);
+
+        _apeStaking.withdrawSelfApeCoin(amount);
+        IERC20(_apeCoin).safeTransfer(msg.sender, amount);
+
+        emit Redeem(msg.sender, amount);
+    }
+
+    function _claim() internal {
+        uint256 pendingYield = yieldAmount(msg.sender);
+        if (pendingYield > 0) {
+            _userPendingYield[msg.sender] = 0;
+            IERC20(_yieldToken).safeTransfer(msg.sender, pendingYield);
+        }
+        emit YieldClaimed(msg.sender, pendingYield);
+    }
+
+    function _harvest(uint160 sqrtPriceLimitX96) internal {
+        //1, get current pending ape coin reward amount
+        uint256 rewardAmount = _apeStaking.pendingRewards(
+            APE_COIN_POOL_ID,
+            address(this),
+            0
+        );
+        if (rewardAmount > 0) {
+            //2, claim pending ape coin reward
+            _apeStaking.claimSelfApeCoin();
+            //3, sell ape coin to usdc
+            uint256 yieldUnderlyingAmount = _sellApeCoinForYieldToken(
+                rewardAmount,
+                sqrtPriceLimitX96
+            );
+            //4, supply usdc to pUsdc
+            IPoolCore(_lendingPool).supply(
+                _yieldUnderlying,
+                yieldUnderlyingAmount,
+                address(this),
+                0
+            );
+            uint256 liquidityIndex = _lendingPool.getReserveNormalizedIncome(
+                _yieldUnderlying
+            );
+            uint256 _yieldAmount = yieldUnderlyingAmount.rayDiv(liquidityIndex);
```

Comment:
_yieldAmount contains liquidityIndex in other places, I think we should be consistent and use `scaled` instead

Reply:
in some other places we call this pendingYied, the naming should better be consistent